elastic/kibana
# [ObsUx] Move the feedback button component to a separate package

Closes #175790
Closes #174347

## Summary

This PR adds `@kbn/feature-feedback-button` which includes the button that was in [observability_shared](https://github.com/elastic/kibana/blob/4bad5ce8cb76a266499ba284618d747aa23de990/x-pack/plugins/observability_shared/public/components/feature_feedback_button/feature_feedback_button.tsx) before. The difference is that I added the possibility to have different query params for Kibana version, environment and sanitized path using a `formConfig` prop. This package replaces also the local ML component. 

## Fixes
Together with the new package creation, this PR fixes the [issue](https://github.com/elastic/kibana/issues/174347) with the K8S form prefilling (kibana version)

## Testing 
- Check the feedback buttons in APM, infra, ML (also in serverless) - some examples:
- Infra:
  -  <img width="1920" alt="image" src="https://github.com/elastic/kibana/assets/14139027/f7b4e7cf-3738-46fc-8e11-a798db140f39">
  - <img width="853" alt="image" src="https://github.com/elastic/kibana/assets/14139027/dda8c81a-f153-48a6-ab2a-306e90223527">
- APM
  - <img width="1920" alt="image" src="https://github.com/elastic/kibana/assets/14139027/492fc74e-4c21-4c62-964c-e0201c12fad8">
    - <img width="853" alt="image" src="https://github.com/elastic/kibana/assets/14139027/dda8c81a-f153-48a6-ab2a-306e90223527"> 
  - <img width="814" alt="image" src="https://github.com/elastic/kibana/assets/14139027/774017a7-117b-402d-81d4-8a8ea49c2bb7">
- ML
  - <img width="1672" alt="image" src="https://github.com/elastic/kibana/assets/14139027/3a63d3c3-b131-461c-8119-32b2f6fb9f2e">

- Check if the k8s button kibana version prefilling is fixed
   - Go to Inventory
   - Select 'Kuberneted Pod' in `Show` menu
   - Click on the feedback button
   - <img width="1666" alt="image" src="https://github.com/elastic/kibana/assets/14139027/ffdf9f57-20ce-4d24-801c-7353a74636b5">
   - <img width="696" alt="image" src="https://github.com/elastic/kibana/assets/14139027/eaeaba02-1638-4a46-a3df-9bb4f1b826d1">
